src/sql/src/ast/transform.rs

```diff
@@ -13,18 +13,121 @@ use std::collections::{BTreeMap, BTreeSet};
+                *schema = Ident::new(new_schema_name.clone());
+            }
+        }
+    };
+
+    match create_stmt {
+        Statement::CreateIndex(CreateIndexStatement { on_name, .. }) => {
+            maybe_update_item_name(on_name.name_mut());
+        }
+        Statement::CreateSink(CreateSinkStatement { from, .. }) => {
+            maybe_update_item_name(from.name_mut());
+        }
+        Statement::CreateTable(CreateTableStatement { name, .. })
+        | Statement::CreateSecret(CreateSecretStatement { name, .. })
+        | Statement::CreateConnection(CreateConnectionStatement { name, .. })
+        | Statement::CreateWebhookSource(CreateWebhookSourceStatement { name, .. })
+        | Statement::CreateType(CreateTypeStatement { name, .. })
+        | Statement::CreateSource(CreateSourceStatement { name, .. }) => {
+            maybe_update_item_name(name);
+        }
+        Statement::CreateView(CreateViewStatement {
+            definition: ViewDefinition { query, name, .. },
+            ..
+        })
+        | Statement::CreateMaterializedView(CreateMaterializedViewStatement {
+            query, name, ..
+        }) => {
+            maybe_update_item_name(name);
+            rewrite_query_schema(cur_schema_name, new_schema_name, query);
+        }
+        _ => unreachable!("Internal error: only catalog items need to update item refs"),
+    }
+}
+
+/// Rewrites `query`'s references of `from` to `to` or errors if too ambiguous.
+fn rewrite_query_schema<'a>(
+    cur_schema_name: &'a str,
+    new_schema_name: String,
+    query: &mut Query<Raw>,
+) {
+    let new_schema = Ident::new(new_schema_name);
+    let mut visitor = CreateSqlRewriteSchema {
+        cur_schema: cur_schema_name,
+        new_schema,
+    };
+    visitor.visit_query_mut(query);
+}
+
+struct CreateSqlRewriteSchema<'a> {
+    cur_schema: &'a str,
+    new_schema: Ident,
+}
+
+impl<'a> CreateSqlRewriteSchema<'a> {
+    fn maybe_rewrite_idents(&mut self, name: &mut [Ident]) {
+        if let [.., schema, _item] = name {
+            if schema.as_str() == self.cur_schema {
+                *schema = self.new_schema.clone();
+            }
+        }
+    }
+}
+
+impl<'a, 'ast> VisitMut<'ast, Raw> for CreateSqlRewriteSchema<'a> {
+    fn visit_expr_mut(&mut self, e: &'ast mut Expr<Raw>) {
+        match e {
+            Expr::Identifier(id) => {
+                // The last ID component is a column name that should not be
+                // considered in the rewrite.
+                let i = id.len() - 1;
+                self.maybe_rewrite_idents(&mut id[..i]);
+            }
+            Expr::QualifiedWildcard(id) => {
+                self.maybe_rewrite_idents(id);
+            }
+            _ => visit_mut::visit_expr_mut(self, e),
+        }
+    }
+    fn visit_unresolved_item_name_mut(
+        &mut self,
+        unresolved_item_name: &'ast mut UnresolvedItemName,
+    ) {
+        self.maybe_rewrite_idents(&mut unresolved_item_name.0);
```

Comment:
uncovered

Reply:
This is surprising! Not sure why this is uncovered